apiserver/worker/backend.py

```diff
@@ -95,6 +101,8 @@ def storeBotRemotely(user_id, bot_id, zip_file_path):
                           },
                           files={"bot.zip": zip_contents})
         print("Posting compiled bot archive %s\n" % r.text)
+        # We don't check the status code since we instaed check the
+        # bot hash.
```

Comment:
There are many error codes that are unlikely to resolve on retry (almost by definition pretty much all the 4xx codes) and we could bail out without trying another 99 times. Also any response error that is likely temporary probably deserves a backoff before retrying.

But leaving it for now is fine, the other changes definitely make it much better than what happens now.